dss_datamover/tests/conftest.py

```diff
@@ -179,3 +304,34 @@ def _method(master):
         master.stop_monitor()
         print("stopping monitoring")
     return _method
+
+class MockMinio():
+    def __init__(self):
+        self.data = {}
+    
+    def list(self, key=''):
+        if not key:
+            return list(self.data.items())
+        else:
+            ret = []
+            for k in self.data.keys():
```

Comment:
the specific key should only show up once. You can check to see if it is a nested dict, if so, just the items() method will pack the dict into a list of tuples

`self.data[k].items() if isinstance(self.data[k], dict) else [(k, self.data[k])]`